tests/integration/targets/virtual_disk/tasks/main.yml

```diff
@@ -0,0 +1,214 @@
+---
+- environment:
+    SC_HOST: "{{ sc_host }}"
+    SC_USERNAME: "{{ sc_username }}"
+    SC_PASSWORD: "{{ sc_password }}"
+    SC_TIMEOUT: "{{ sc_timeout }}"
+
+# ----------------------------------Setup------------------------------------------------------------------------
+  block:
+    - name: Delete virtual disk files if exist
+      scale_computing.hypercore.virtual_disk:
+        state: absent
+        file_name: "{{ item }}"
+      loop:
+        - xlab-ci-test-VD.qcow2
+        - xlab-ci-test-VD.vmdk
+
+    - name: Generate virtual disk file qcow2
+      ansible.builtin.shell:
+        cmd: qemu-img create -f qcow2 xlab-ci-test-VD.qcow2 10M
+      register: generated_img
+
+    - name: Generate virtual disk file vmdk
+      ansible.builtin.shell:
+        cmd: qemu-img create -f vmdk xlab-ci-test-VD.vmdk 10M
+      register: generated_img
+
+# ----------------------------------Job------------------------------------------------------------------------
+    - name: Upload virtual disk file .qcow2
+      scale_computing.hypercore.virtual_disk:
+        state: present
+        file_location: xlab-ci-test-VD.qcow2
+        file_name: xlab-ci-test-VD.qcow2
+      register: uploaded_virtual_disk
+    - ansible.builtin.assert:
+        that:
+          - uploaded_virtual_disk is succeeded
+          - uploaded_virtual_disk is changed
+          - uploaded_virtual_disk.record.name == "xlab-ci-test-VD.qcow2"
```

Comment:
Check also `uploaded_virtual_disk.record.keys() == [...]` is as expected, and also their values.